src/app/app.component.ts

```diff
@@ -1,10 +1,92 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { Item } from './models/item';
+import { GroceryService } from './grocery.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent {
+export class AppComponent implements OnInit{
+  groceries: Item[] = [];
+  displayedGroceries: Item[] = [];
   title = 'angular-green-grocers';
+  cart: { item: Item, quantity: number }[] = [];
+  selectedType: string = '';
+  
+  constructor(private groceryService: GroceryService) { }
+
+
+  ngOnInit() {
+    this.getGroceries();
+  }
+
+  getGroceries() {
+    this.groceryService.getGroceries().subscribe({
```

Comment:
Remember we need to store the subscriptions, and unsubscribe when we're done -  otherwise we'll create a memory leak